spring-cloud-gcp-data-firestore/src/main/java/com/google/cloud/firestore/PublicClassMapper.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright 2019-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.firestore;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import com.google.cloud.Timestamp;
+import com.google.firestore.v1.Document;
+import com.google.firestore.v1.Value;
+
+/**
+ * @author Dmitry Solomakha
+ */
+public final class PublicClassMapper {
```

Comment:
Did you copy-paste logic into this class, or is all of this missing in the firestore library?
I expected to see a simple wrapper, but that's not what this looks like.